cloudinit/config/cc_raspberry_pi.py

```diff
@@ -115,9 +111,20 @@ def get_bool_field(cfg_dict: dict, name: str, default=False):
         except subp.ProcessExecutionError as e:
             LOG.error("Failed to configure serial hardware: %s", e)
 
-        require_reboot(instCfg)
+        # Reboot to apply changes
+        cmd = cloud.distro.shutdown_command(
+            mode="reboot",
+            delay="now",
+            message="Rebooting to apply serial console changes",
+        )
+        proc = subprocess.Popen(
+            cmd, stdin=subprocess.PIPE, stdout=None, stderr=subprocess.STDOUT
+        )
+        ret = proc.returncode
     except subp.ProcessExecutionError as e:
         LOG.error("Failed to configure serial console: %s", e)
+        os._exit(254)
```

Comment:
we shouldn't be exitting manually on either of these next two lines. Cloud-init has a separate signal handler for dealing being killed due to reboots.

Reply:
Got it.